modules/porous_flow/src/kernels/PorousFlowFullySaturatedMassTimeDerivative.C

```diff
@@ -0,0 +1,100 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowFullySaturatedMassTimeDerivative.h"
+
+template<>
+InputParameters validParams<PorousFlowFullySaturatedMassTimeDerivative>()
+{
+  InputParameters params = validParams<TimeKernel>();
+  MooseEnum simulation_type("Hydro=0 ThermoHydro=1 HydroMechanical=2 ThermoHydroMechanical=3", "Hydro");
+  params.addParam<MooseEnum>("simulation_type", simulation_type, "The type of simulation.  For simulations involving Mechanical deformations, you will need to supply the correct Biot coefficient.  For simulations involving Thermal flows, you will need an associated ConstantThermalExpansionCoefficient Material");
+  params.addRangeCheckedParam<Real>("biot_coefficient", 1.0, "biot_coefficient>=0 & biot_coefficient<=1", "Biot coefficient");
+  params.addParam<bool>("multiply_by_density", true, "If true, then this Kernel is the time derivative of the fluid mass.  If false, then this Kernel is the derivative of the fluid volume (which is common in poro-mechanics)");
+  params.addRequiredParam<UserObjectName>("PorousFlowDictator", "The UserObject that holds the list of Porous-Flow variable names.");
+  params.addClassDescription("Fully-saturated version of the single-component, single-phase fluid mass derivative wrt time");
+  return params;
+}
+
+PorousFlowFullySaturatedMassTimeDerivative::PorousFlowFullySaturatedMassTimeDerivative(const InputParameters & parameters) :
+    TimeKernel(parameters),
+    _dictator(getUserObject<PorousFlowDictator>("PorousFlowDictator")),
+    _var_is_porflow_var(_dictator.isPorousFlowVariable(_var.number())),
+    _multiply_by_density(getParam<bool>("multiply_by_density")),
+    _simulation_type(getParam<MooseEnum>("simulation_type")),
```

Comment:
If you define a C++ enum that corresponds to your MOOSE enum. You can initialize it in a one-liner just like you do a normal MOOSE enum, but then you can use the named constants in the rest of your code to avoid comparing against numbers like you do in the lines below.